ClientRuntimes/NodeJS/ms-rest/lib/serialization.js

```diff
@@ -266,8 +266,8 @@ function serializeCompositeType(mapper, object, objectName) {
           continue;
         }
         //serialize the property if it is present in the provided object instance
-        if ((modelProps[key].defaultValue !== null && modelProps[key].defaultValue !== undefined) || 
-          (object[key] !== null && object[key] !== undefined)) {
+        if (((parentObject !== null && parentObject !== undefined) && (modelProps[key].defaultValue !== null && modelProps[key].defaultValue !== undefined)) || 
+            (object[key] !== null && object[key] !== undefined)) {
           var propertyObjectName = objectName + '.' + modelProps[key].serializedName;
```

Comment:
@stankovski - Can you verify this node.js fix for issue #782?
So, if the parentObject's mapper states that it is optional and if it is not provided in the actual object to be serialized then the parentObject is undefined. In that case we will not serialize the parentObject. Previously we used to satisfy the if condition and then the serialization would fail on line 274 where it tries to assign the serialized constant value to its **undefined** parentObject. Now, this will not happen.

Just trying to give you enough context so you can make sure, I don't miss any edge case.